### PR TITLE
refactor the get_headers

### PR DESCRIPTION
**Pull request by g302ge**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number:  3391

## What is the new behavior?

Refactor the chain_service `get_headers` function return type wrapped by Option. Currently the `get_headers` will return an Option wrapper directly.

### CODE REVIEW

**Comment by nkysg** [comment]
@g302ge Hi, please check this script https://github.com/starcoinorg/starcoin/blob/master/scripts/check_commit.sh and retry。

File: rpc/server/src/module/chain_rpc.rs [hunk 1]
```diff
@@ -377,7 +377,7 @@ where
         let service = self.service.clone();
         let fut = async move {
             let headers = service.get_headers(block_hashes).await?;
-            Ok(headers.into_iter().map(Into::into).collect())
+            Ok(headers.into_iter().flatten().map(Into::into).collect())
```
Comment (nkysg) [inline rpc/server/src/module/chain_rpc.rs:380]: 这里不用flattern吧，flatten None的header会略过吧

File: chain/service/src/chain_service.rs [hunk 2]
```diff
@@ -302,14 +302,12 @@ impl ReadableChainService for ChainReaderServiceInner {
     }
 
     // XXX FIXME different get_blocks
```
[the remark this answers is not in the document]
Comment (g302ge) [reply chain/service/src/chain_service.rs:304]: OK

File: rpc/server/src/module/chain_rpc.rs (hunk 1 above)
Comment (g302ge) [reply rpc/server/src/module/chain_rpc.rs:380]: 这里是拆开包装，然后直接返回回去原来的结构

Comment (g302ge) [reply rpc/server/src/module/chain_rpc.rs:380]: 看起来是我们需要把这个返回结果也得安排 Option 回去

Comment (nkysg) [reply rpc/server/src/module/chain_rpc.rs:380]: 我看错了，应该没有问题

**Review by nkysg — APPROVED** [review]

**Comment by nkysg** [comment]
https://github.com/starcoinorg/starcoin/issues/3391

**Pull request merged**